pandas/tests/test_nanops.py

```diff
@@ -623,124 +798,137 @@ def test_nancov(self):
-
-            # Lower dimension for next step in the loop
-            arr_float = np.take(arr_float, 0, axis=-1)
-            arr_float1 = np.take(arr_float1, 0, axis=-1)
-            arr_nan = np.take(arr_nan, 0, axis=-1)
-            arr_nan_nan = np.take(arr_nan_nan, 0, axis=-1)
-            arr_float_nan = np.take(arr_float_nan, 0, axis=-1)
-            arr_float1_nan = np.take(arr_float1_nan, 0, axis=-1)
-            arr_nan_float1 = np.take(arr_nan_float1, 0, axis=-1)
-            targ0 = np.take(targ0, 0, axis=-1)
 
-    @pytest.mark.parametrize(
-        "arr, correct",
-        [
-            ("arr_complex", False),
-            ("arr_int", False),
-            ("arr_bool", False),
-            ("arr_str", False),
-            ("arr_utf", False),
-            ("arr_complex", False),
-            ("arr_complex_nan", False),
-            ("arr_nan_nanj", False),
-            ("arr_nan_infj", True),
-            ("arr_complex_nan_infj", True),
-        ],
-    )
-    def test__has_infs_non_float(self, arr, correct):
-        val = getattr(self, arr)
-        while getattr(val, "ndim", True):
-            res0 = nanops._has_infs(val)
-            if correct:
-                assert res0
-            else:
-                assert not res0
+@pytest.mark.parametrize(
+    "op,nanop",
+    [
+        (operator.eq, nanops.naneq),
+        (operator.ne, nanops.nanne),
+        (operator.gt, nanops.nangt),
+        (operator.ge, nanops.nange),
+        (operator.lt, nanops.nanlt),
+        (operator.le, nanops.nanle),
+    ],
+)
+def test_nan_comparison(request, op, nanop, disable_bottleneck):
+    arr_float = request.getfixturevalue("arr_float")
+    arr_float1 = request.getfixturevalue("arr_float1")
+    targ0 = op(arr_float, arr_float1)
+    arr_nan = request.getfixturevalue("arr_nan")
+    arr_nan_nan = request.getfixturevalue("arr_nan_nan")
+    arr_float_nan = request.getfixturevalue("arr_float_nan")
+    arr_float1_nan = request.getfixturevalue("arr_float1_nan")
+    arr_nan_float1 = request.getfixturevalue("arr_nan_float1")
+
+    while targ0.ndim:
+        res0 = nanop(arr_float, arr_float1)
+        tm.assert_almost_equal(targ0, res0)
+
+        if targ0.ndim > 1:
+            targ1 = np.vstack([targ0, arr_nan])
+        else:
+            targ1 = np.hstack([targ0, arr_nan])
+        res1 = nanop(arr_float_nan, arr_float1_nan)
+        tm.assert_numpy_array_equal(targ1, res1, check_dtype=False)
+
+        targ2 = arr_nan_nan
+        res2 = nanop(arr_float_nan, arr_nan_float1)
+        tm.assert_numpy_array_equal(targ2, res2, check_dtype=False)
+
+        # Lower dimension for next step in the loop
+        arr_float = np.take(arr_float, 0, axis=-1)
+        arr_float1 = np.take(arr_float1, 0, axis=-1)
+        arr_nan = np.take(arr_nan, 0, axis=-1)
+        arr_nan_nan = np.take(arr_nan_nan, 0, axis=-1)
+        arr_float_nan = np.take(arr_float_nan, 0, axis=-1)
+        arr_float1_nan = np.take(arr_float1_nan, 0, axis=-1)
```

Comment:
Can we remove one of both lines? IIUC are_float_nan and arr_float1_nan have the same content

Reply:
I think these need to be separate objects for the test, but I was able to instantiate 2 separate objects with the same fixture